nomad/data_source_acl_tokens.go

```diff
@@ -0,0 +1,82 @@
+package nomad
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceACLTokens() *schema.Resource {
+	return &schema.Resource{
+		Read: aclTokensDataSourceRead,
+
+		Schema: map[string]*schema.Schema{
+			"prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"acl_tokens": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"accessor_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"policies": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"global": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"create_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func aclTokensDataSourceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(ProviderConfig).client
+
+	qOpts := &api.QueryOptions{
+		Prefix: d.Get("prefix").(string),
+	}
+	tokens, _, err := client.ACLTokens().List(qOpts)
+	if err != nil {
+		return fmt.Errorf("error while getting the list of tokens: %v", err)
+	}
+
+	result := make([]map[string]interface{}, len(tokens))
+	for i, t := range tokens {
+		result[i] = map[string]interface{}{
+			"accessor_id": t.AccessorID,
+			"name":        t.Name,
+			"type":        t.Type,
+			"policies":    t.Policies,
+			"global":      t.Global,
+			"create_time": t.CreateTime.String(),
+		}
+	}
+
+	d.SetId("nomad-tokens")
```

Comment:
apologies for not noticing this before... does this ID need to include the prefix? different `data "nomad_acl_tokens"` with different prefix options should (probably?) have different IDs.

Reply:
i manually tested an terraform didn't complain... i will ask someone from the provider team on Monday.

Reply:
I think the ID does not matters as long as it is not `""` as this is used to indicate, it's usually set to the external ID for resources that have one but when a Terraform resource is a virtual resource (like https://github.com/hashicorp/terraform-provider-null/blob/master/null/data_source.go#L51) or is actually multiple resource it can be set to something else.

I can push a commit to change it to `d.SetId(client.Address() + "/tokens/" + prefix)` which would make it more like the pattern used by the other datasources if you want.